code/obj/item/device/pda2/bot_control.dm

```diff
@@ -146,6 +146,127 @@
 
 		return
 
+/datum/computer/file/pda_program/bot_control/secbotpro
```

Comment:
Seems like big parts of this are copypaste from the regular secbot program. I think it wouldn't be too hard to make this a subtype and have it just add the relevant parts in return_text and Topic.

Reply:
Done, now its slightly smaller and significantly harder to read.